reporter.go

```diff
@@ -22,8 +22,15 @@ package tally
 
 import "time"
 
+type baseStatsReporter interface {
```

Comment:
Might be worth making this a public interface on second thoughts since its strange to export a public interface that extends a privately defined interface.